next-cloudinary/src/components/CldUploadWidget/CldUploadWidget.types.ts

```diff
@@ -8,6 +8,39 @@ export interface CldUploadWidgetResults {
   info?: string | object;
 }
 
+type CldUploadWidgetDetsroyInstanceMethodOptions = {
+  removeThumbnails?: boolean;
+}
+
+type CldUploadWidgetCloseInstanceMethodOptions = {
+  quiet?: boolean;
+}
+
+type CldUploadWidgetOpenInstanceMethodOptions = {
+  files?: URL[];
+}
+
+type CldUploadWidgetUpdateInstanceMethodOptions = Omit<
+  CldUploadWidgetPropsOptions,
+  "secure" | "uploadSignature" | "getTags" | "preBatch" | "inlineContainer" | "fieldName"
+> & {
+  cloudName: string;
+  uploadPreset: string;
+}
+
+export interface CldUploadWidgetInstanceMethods {
+  close: (options?: CldUploadWidgetCloseInstanceMethodOptions) => void;
```

Comment:
one thing to consider with how you're adding these is it changes the API for how the original Upload Widget expects the arugment to be passed in

```
.close(true)
```
vs
```
.close({
  quiet: true
})
```

i'm not necessarily opposed to the latter, which is how you have it, though its worth being mindful of that change

one benefit of arguments "as is" is that the cloudinary docs is easily referenceable for the API

one benefit of arguments as an object is its likely a bit easier to manage for more complex instance methods as well as the potential to expand on the API should there be any interesting use cases that could enhance the experience, which might make more sense for some of the others methods available

Reply:
Yes, I agree with both of your points. But I think there is some issues in the docs, If you notice there is difference between the syntax shown and the example given. For example for `close` method:

[Example provided in doc](https://cloudinary.com/documentation/upload_widget_reference#example-6):
```
var widget = cloudinary.createUploadWidget({ 
  cloudName: "my-cloud-name", uploadPreset: "preset1"}, (error, result) => { });
widget.close({quiet: true});
```
[Syntax](https://cloudinary.com/documentation/upload_widget_reference#syntax-5):
```
close(quiet)
```

So some clarification is needed here on what approach do we want to use and update the same in the docs as well (both `syntax` & `example`) accordingly to avoid confusions.

Reply:
oops apologies, that was my misunderstanding of how that was being presented on the docs page!

`open` currently has an error, i have a ticket for that to be addressed by the docs team. the 2nd argument takes an `options` object with `files` as an available property

it also appears this is a similar situation with both `close` and `destroy`. im trying to confirm which syntax is right, but also opening tickets for that to be addressed in the docs

further, it sounds like `null` can actually be a value related to the tab that's opened in the widget UI, but i need to get clarification and guidance around that, ill work on getting that for you

Reply:
okay found out about the first value. this isn't documented yet, but i learned that this can be a string at a minimum which represents the source that the person wants to open in the widget

<img width="853" alt="image" src="https://github.com/colbyfayock/next-cloudinary/assets/1045274/28733a7c-49b7-4b92-a1bf-d316edfe01ff">

for instance, passing `.open('url')` will open the URL tab

a list of these sources are made available on the `cloudinary` instance

<img width="619" alt="image" src="https://github.com/colbyfayock/next-cloudinary/assets/1045274/9e4816a4-cdca-4b57-b256-a94926775905">

the 2nd argument is definitely an object, so some examples here:

```
open('url') // open(cloudinary.WIDGET_SOURCES.URL);

open(null, {
  files: ['https://image.com/image.jpg']
});
```

Reply:
ive also confirmed in testing that the close method is an object with quiet as a property

```
widget.close({
  quiet: true
});
```

Reply:
Got it! Also, for `open` does both params are optional? as [here](https://github.com/colbyfayock/next-cloudinary/blob/60c4b7c895056e2abf0a0f7aec54b0a3bdd04cb8/next-cloudinary/src/components/CldUploadButton/CldUploadButton.tsx#L63) its getting called without any argument.

Reply:
Yup both optional!